packages/examples/src/index.ts

```diff
@@ -0,0 +1,52 @@
+// Sol
+import bs58 from "bs58";
+import { Keypair } from "@solana/web3.js";
+// Eth
+import { ethers } from "ethers";
+
+import * as wh from "@wormhole-foundation/sdk-base";
+import * as sol from "@wormhole-foundation/sdk-solana";
+import * as evm from "@wormhole-foundation/sdk-evm";
+//import * as sui from '@wormhole-foundation/sdk-sui';
+
+// Please don't steal my testnet moneys
+function getSolanaSigner(): Keypair {
+  return Keypair.fromSecretKey(
+    bs58.decode(
+      "2v5fKQHaDLuWYBQCzFGvovnRNXPy8jkErWkeSMigs1PdsnQvRC5EMX3CJdULaEaQqaMNagfhsoj8sfQ7Dn2MnjKy"
+    )
+  );
+}
+
+function getEthSigner(provider: ethers.providers.Provider): ethers.Wallet {
+  const ethPk =
+    "0x3f493e59e81db1be4ebbe18b28ba8fdd066ef44139420ead59f37f5dacb80719";
+  return new ethers.Wallet(ethPk, provider);
+}
+
+(async function () {
+  const w = new wh.Wormhole(wh.Network.TESTNET, {
+    [wh.Context.SOLANA]: new sol.SolanaContext(wh.Network.TESTNET),
+    [wh.Context.EVM]: new evm.EvmContext(wh.Network.TESTNET),
+    // complains about some dep required by mysten
+    //[wh.Context.SUI]: new sui.SuiContext(wh.Network.TESTNET),
+  });
+
+  const ethSigner = getEthSigner(w.mustGetProvider("goerli"));
+  const solSigner = getSolanaSigner();
+
+  const tx = await w.startTransfer(
+    "native", // TokenId
+    10000000n, // Amount in base units
+
+    // can we pass an interface that provides
+    // both the chain and a fn to get the address?
+    "solana",
+    solSigner.publicKey.toBase58(),
```

Comment:
Maybe we offer a `Signer<Context>` type class that wraps chain native wallet stuff and provides `getChain`, `getAddress`, and `sign`?